src/libPMacc/include/memory/buffers/DeviceBufferIntern.hpp

```diff
@@ -56,14 +56,14 @@ class DeviceBufferIntern : public DeviceBuffer<TYPE, DIM>
         //create size on device before any use of setCurrentSize
         if (useVectorAsBase)
         {
-            sizeOnDevice = false;
-            createSizeOnDevice(sizeOnDevice);
+            this->sizeOnDevice = false;
+            createSizeOnDevice(this->sizeOnDevice);
             createFakeData();
             this->data1D = true;
         }
         else
         {
-            createSizeOnDevice(sizeOnDevice);
```

Comment:
QUESTION:
This means that the constructor argument `sizeOnDevice` is always ignored?
Is this the wanted behavior?

Why is then in the following constructor the constructor argument used but not the member variable  `sizeOnDevice`.